src/python/dxpy/scripts/dx.py

```diff
@@ -3446,7 +3438,7 @@ def register_subparser(subparser, subparsers_action=None, categories=('other', )
                                         description='Clears all environment variables set by dx.  More specifically, it removes local state stored in ~/.dnanexus_config/environment.  Does not affect the environment variables currently set in your shell.', prog='dx clearenv')
 parser_clearenv.add_argument('--reset', help='Reset dx environment variables to empty values. Use this to avoid interference between multiple dx sessions when using shell environment variables.',
                              action='store_true')
-parser_clearenv.set_defaults(func=clearenv, interactive=True)
+parser_clearenv.set_defaults(func=clearenv, interactive=False)
```

Comment:
Previously, `dx clearenv` would invoke a hardcoded shortcut that would avoid initializing dxpy from the environment. The invocation through the argparse parser would only happen when running `dx sh`. Now we can use the parser for both.